tiledb/cloud/array.py

```diff
@@ -1,13 +1,49 @@
-import urllib
 from . import rest_api
 from . import config
 from . import client
-from . import tasks
 from . import cloudarray
 from . import tiledb_cloud_error
 from .rest_api import ApiException as GenApiException
 from .rest_api import rest
 
+import zlib
+import multiprocessing
+import cloudpickle
+
+
+class UDFResult(multiprocessing.pool.ApplyResult):
+    def __init__(self, response):
+        self.response = response
+
+    def get(self, timeout=None):
+        try:
+            response = rest.RESTResponse(self.response.get(timeout=timeout))
+
+            global last_udf_task_id
+            last_udf_task_id = response.getheader(client.TASK_ID_HEADER)
+
+            res = response.data
+
+        except GenApiException as exc:
```

Comment:
`response.get` will raise `multiprocessing.TimeoutError` if it fails, so need to check that here as well.